OpenRA.Mods.Common/Activities/FindResources.cs

```diff
@@ -66,39 +66,43 @@ public override Activity Tick(Actor self)
 			if (!closestHarvestablePosition.HasValue)
 			{
 				if (!harv.IsEmpty)
-					return deliver;
+					return ActivityUtils.SequenceActivities(deliver, this);
```

Comment:
This change is the cause of @obrakmann's infinite loop.  DeliverResources has a check at the start of its tick to do nothing and return the next activity is that is non-null, so this will always loop.